app/react/src/client/preview/render.tsx

```diff
@@ -28,9 +37,50 @@ export const render: ArgsStoryFn<ReactFramework> = (args, context) => {
 
 const renderElement = async (node: ReactElement, el: Element) =>
   new Promise((resolve) => {
-    ReactDOM.render(node, el, () => resolve(null));
+    // Create Root Element conditionally for new React 18 Root Api
+    const root = getReactRoot(el);
+
+    if (root) {
+      root.render(node);
+      setTimeout(() => {
+        resolve(null);
+      }, 0);
+    } else {
+      ReactDOM.render(node, el, () => resolve(null));
+    }
   });
 
+const unmountElement = (el: Element) => {
+  const root = nodes.get(el);
+  if (root && FRAMEWORK_OPTIONS?.newRootApi) {
+    root.unmount();
+    nodes.delete(el);
+  } else {
+    ReactDOM.unmountComponentAtNode(el);
+  }
+};
+
+const getReactRoot = (el: Element): IRoot | null => {
+  if (!FRAMEWORK_OPTIONS?.newRootApi) {
+    return null;
+  }
+
+  if (!(ReactDOM as any).createRoot) {
```

Comment:
This is now throwing on `react-dom@18.0.0-rc.2` since `createRoot` is only available if it's imported as `import ReactDOM from 'react-dom/client`.

Announcement here: https://github.com/reactwg/react-18/discussions/5

Reply:
Might be worth noting how Nextjs checks whether it should use React18 apis or not: https://github.com/vercel/next.js/blob/9bff48b9d5998455c9c038d7f5834a58bb9f429f/packages/next/server/config.ts#L736

Reply:
Thanks @valentinpalkovic for your persistence and patience here. I'll also spend some time on it this weekend and hope we can get it merged early next week. 🙏